services/wallet/walletconnect/service.go

```diff
@@ -0,0 +1,53 @@
+package walletconnect
+
+import (
+	"github.com/ethereum/go-ethereum/event"
+	"github.com/status-im/status-go/multiaccounts/accounts"
+	"github.com/status-im/status-go/rpc/network"
+)
+
+type Service struct {
+	networkManager *network.Manager
+	accountsDB     *accounts.Database
+	eventFeed      *event.Feed
+}
+
+func NewService(networkManager *network.Manager, accountsDB *accounts.Database, eventFeed *event.Feed) *Service {
+	return &Service{
+		networkManager: networkManager,
+		accountsDB:     accountsDB,
+		eventFeed:      eventFeed,
+	}
+}
+
+func (s *Service) PairSessionProposal(proposal SessionProposal) (*PairSessionResponse, error) {
+	namespace := Namespace{
+		Methods: []string{"eth_sendTransaction", "personal_sign"},
+		Events:  []string{"accountsChanged", "chainChanged"},
+	}
+
+	proposedChains := proposal.Params.RequiredNamespaces.Eip155.Chains
+	chains, eipChains := sessionProposalToSupportedChain(proposedChains, func(chainID uint64) bool {
+		return s.networkManager.Find(chainID) != nil
+	})
+	if len(chains) != len(proposedChains) {
+		return nil, ErrorChainsNotSupported
+	}
+	namespace.Chains = eipChains
+
+	activeAccounts, err := s.accountsDB.GetActiveAccounts()
+	if err != nil {
+		return nil, ErrorChainsNotSupported
```

Comment:
I don't imagine we'll get this error often, but we could generate a new one instead of `ErrorChainsNotSupported` since it's not really related

Reply:
Good point. I already fixed it in the following PR: https://github.com/status-im/status-go/blob/5ab9e8e4e7daa5f45a9f5db57a3c9e42889af04b/services/wallet/walletconnect/service.go#L52